scipy/signal/tests/test_signaltools.py

```diff
@@ -652,7 +652,24 @@ def test_basic(self):
         # make sure interpolated values are on knot points
         assert_array_almost_equal(y2[::10], y, decimal=5)
 
-
+    def test_complex(self):
+        # create some smoothly varying complex signal to try and interpolate
+        x = np.arange(100)
+        y = np.zeros(x.shape, dtype=np.complex64)
+        T = 10.0
+        f = 1.0 / T
+        y = np.sin(2.0 * np.pi * f * x) + \
```

Comment:
You're correct. When I originally wrote the test I was trying to verify what was going on with pure real or complex terms so I wrote it up to easily add/remove the imaginary terms.

For the test purposes it does not matter what the input data is, as long as it comes across as complex before the interpolation. 

The test code also computes many more values than are necessary to perform the test. I could cut down the test to save computation if that is seen as important.